src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +45,55 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+
+        if (getDate(dateFrom).isAfter(getDate(dateTo))) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        DateTimeFormatter sdf = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        StringBuilder result = new StringBuilder("Отчёт за период ")
+                .append(sdf.format(getDate(dateFrom)))
+                .append(" - ")
+                .append(sdf.format(getDate(dateTo)));
+
+        int salary = 0;
+        for (String name : names) {
+            for (String currentData : data) {
+                if ((name.equals(getName(currentData)))
+                        & (dateCondition(currentData, dateFrom, dateTo))) {
+                    salary += getHours(currentData) * getPrice(currentData);
+                }
+            }
+            result.append("\n").append(name).append(" - ").append(salary);
+            salary = 0;
+        }
+        return result.toString();
+    }
+
+    private boolean dateCondition(String current, String dateFrom, String dateTo) {
+        return (getDate(current).isAfter(getDate(dateFrom))
```

Comment:
```suggestion
        return getDate(current).isAfter(getDate(dateFrom))
```
inital braces are redundant